### PR TITLE
Fix some LSP bugs causing code actions to not show up correctly

### PR DESCRIPTION
**Pull request by TudbuT**
- [x] Added an entry to `CHANGELOG.md` if this change could be valuable to users

This will fix the following LSP-related issues: 
- Java LS does not show "Surround with try/catch" and similar issue-related actions
- All language servers don't always update code actions when the document was changed
- Rust LS sometimes highlights a line with a fix suggestion, but is unable to actually provide it in code actions

This PR updates the capabilities and code action requests to fully match LSP spec.

### CODE REVIEW

**Comment by TudbuT** [comment]
Wait what, why'd clippy fail here but not on my local PC?

File: lapce-proxy/src/plugin/mod.rs [hunk 1]
```diff
@@ -488,7 +489,16 @@ impl PluginCatalogRpcHandler {
                 start: position,
                 end: position,
             },
-            context: CodeActionContext::default(),
+            context: CodeActionContext {
+                diagnostics: diagnostics
+                    .into_iter()
+                    .filter(|x| {
```
Comment (dzhou121) [inline lapce-proxy/src/plugin/mod.rs:495]: Maybe it's better to filter the diagnostics in the code actions request.

Comment (TudbuT) [reply lapce-proxy/src/plugin/mod.rs]: Done

File: lapce-data/src/editor.rs [hunk 2]
```diff
@@ -288,34 +288,50 @@ impl LapceEditorBufferData {
...
-                                    resp,
-                                },
-                                Target::Auto,
-                            );
-                        }
-                    },
-                );
-            }
+            // REMOVED(if) this prevents some code actions from showing up correctly, and there is almost no performance penalty for this
+            //if self.doc.code_actions.get(&prev_offset).is_none() {
```
Comment (dzhou121) [inline lapce-data/src/editor.rs:292]: Does this make it not work properly?

Comment (TudbuT) [reply lapce-data/src/editor.rs:292]: The if-statement causes some delayed code actions to be left out, causing some quick fixes to not appear. REMOVED(if) is just the reason why it was removed and what was removed, in this case an if statement.

Comment (dzhou121) [reply lapce-data/src/editor.rs:292]: Is there any way I can test the delayed code actions bit?

Comment (TudbuT) [reply lapce-data/src/editor.rs:292]: Take this rust code:

```rs
fn main() {
    let x = 0;
    x = 1;
    println!(x);
}
```

There are two errors here. Open quick fixes on the variable name, there will be no fix. Now you fix the println one, which has a higher internal priority in RA. Save the file. The bug happens when you open quick fixes on the variable name again, even after you have now fixed the println one and it shows a new error.

Comment (TudbuT) [reply lapce-data/src/editor.rs:292]: The same goes for code actions on already-modified code.

**Pull request merged**